### ISSUE TITLE
implement PRNG functionality

### ISSUE THREAD

**Issue by aupiff** [aupiff/keccak#2]
https://keccak.team/files/SpongePRNG.pdf

**Comment by aupiff** [comment]
consider the PRNG interfaces used in this library:

https://hackage.haskell.org/package/DRBG-0.5.5/docs/Crypto-Random-DRBG.html

read:
- https://en.wikipedia.org/wiki/Pseudorandom_number_generator
- http://www.firstpr.com.au/dsp/rand31/

**Comment by pindaroso** [comment]
What's the timeline for getting this done, your goal? Still need to do some research on PRNG, but I'm super curious, interest piqued. If you plan on having this done in 2-3 weeks, happy to claim.

**Comment by aupiff** [comment]
Before I start on this I need to expand the test suite (sha3 and keccak), work with long inputs to the hash (longer than a block), support the standard test suites mentioned in #3, and get Travis running my tests. I'll probably have this done in 2 weeks after which you or I could work on the PRNG functionality. If you want to help me with basic keccak hashing functionality & testing, I'd appreciate it!

**Comment by pindaroso** [comment]
Gotcha. I’ll start digging in. Thanks for the info!

**Comment by aupiff** [comment]
Extensions of SHA-3, e.g. SHAKE, ParallelHash, etc:

https://nvlpubs.nist.gov/nistpubs/SpecialPublications/NIST.SP.800-185.pdf

**Comment by aupiff** [comment]
I'm adding SHAKE in a PR later today along with the corresponding KATs. The PR will resolve this issue. A release of version 0.2.0 of this library will happen once SHAKE is merged into master and I've finished with most of my planned optimizations.
